plugins/inputs/logfile/tailersrc_test.go

```diff
@@ -321,7 +321,7 @@ func TestTailerSrcFiltersSingleLineLogs(t *testing.T) {
 	n := 100
 	matchedLog := "ERROR: this has an error in it."
 	unmatchedLog := "Some other log message"
-	publishLogsToFile(resources.file, matchedLog, unmatchedLog, n, 0)
+	publishLogsToFile(resources.file, matchedLog, unmatchedLog, n, 100)
```

Comment:
1, the sleep timer that is in the test + the multilineWaitPeriod interval in the tailersrc implementation should not impact ingestion of log lines when there is no multiline function (the tailer created in this test passes in `nil` for that in line 318). The wait period is used to determine how long the agent waits with no new logs ingested before it pushes what is in memory and flushes. 

2, I already have my eye on these tests to figure out why they are flaky - I also wrote them.

Reply:
Makes sense on 1, and I now realise my most recent change is pointless because leaving it at `0` won't update `multilineWaitPeriod` anyway and thus they'll never be a race condition there.

On 2, many thanks for looking into it! I suspect you'll have much more luck figuring this out than I will

Reply:
#340 should hopefully address the flakiness in these tests. Can you revert this change that you made?

Reply:
Done